src/full/Agda/Interaction/Library.hs

```diff
@@ -177,42 +177,41 @@ findProjectConfig root = mkLibM [] $ findProjectConfig' root
 findProjectConfig'
   :: FilePath                          -- ^ Candidate (init: the directory Agda was called in)
   -> LibErrorIO ProjectConfig          -- ^ Actual root and @.agda-lib@ files for this project
-findProjectConfig' root = do
-  getCachedProjectConfig root >>= \case
-    Just conf -> return conf
-    Nothing   -> do
-      libFiles <- liftIO $ filter ((== ".agda-lib") . takeExtension) <$> getDirectoryContents root
-      case libFiles of
-        []     -> liftIO (upPath root) >>= \case
-          Just up -> do
-            conf <- findProjectConfig' up
-            storeCachedProjectConfig root conf
-            return conf
-          Nothing -> return DefaultProjectConfig
-        files -> do
-          let conf = ProjectConfig root files
-          storeCachedProjectConfig root conf
-          return conf
-
+findProjectConfig' dir =
+  search . parents =<< liftIO (makeAbsolute dir)
   where
-    -- Note that "going up" one directory is OS dependent
```

Comment:
Please retain this comment in some form, as it documents a design path we have taken in the past.  Future developers might come back to this code and wonder why we do what we do, and preserving a discussion of past choices is enlightening.

Reply:
Sure, if we decide to switch to the new semantics.